elasticsearch/connection/http_requests.py

```diff
@@ -80,6 +82,8 @@ def __init__(
             elif isinstance(http_auth, string_types):
                 http_auth = tuple(http_auth.split(":", 1))
             self.session.auth = http_auth
+        if api_key is not None:
+            self.session.headers.setdefault('authorization', self._get_api_key_header_val(api_key))
```

Comment:
I don't think we should be using `setdefault` here as it might be confusing when the `api_key` parameter will be ignored if `authorization` is already in `headers`.

Either override the value (since it is an explicit request of the user to pass `api_key`) or raise an exception if there already is an `authorization` header present.

Reply:
I opted to override the `authorization` header if `api_key` is set.